whipper/common/program.py

```diff
@@ -470,6 +471,35 @@ def getHTOA(self):
         stop = track.getIndex(1).absolute - 1
         return start, stop
 
+    def getCoverArt(self, path, release_id):
+        """
+        Get cover art image from Cover Art Archive.
+
+        :param path: where to store the fetched image
+        :type  path: str
+        :param release_id: a release id (self.program.metadata.mbid)
+        :type  release_id: str
+        :returns: path to the downloaded cover art, else `None`
+        :rtype: str or None
+        """
+        cover_art_path = os.path.join(path, 'cover.jpg')
+
+        logger.debug('fetching cover art for release: %r', release_id)
+        try:
+            data = musicbrainzngs.get_image_front(release_id, 500)
+        except musicbrainzngs.ResponseError as e:
+            logger.error('error fetching cover art: %r', e)
+            return
+
+        if data:
+            fd, tmppath = tempfile.mkstemp(suffix='.cover.jpg')
+            with os.fdopen(fd, 'wb') as f:
```

Comment:
Instead of doing these two lines, it might be better to use [`tempfile.TemporaryFile`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryFile) or one of the other methods that allow to use `with` directly with the temporary file, so that we’re sure the file gets cleaned up once we’re done with it.

See e.g., https://stackoverflow.com/questions/18458408/how-to-call-tempfile-mkstemp-with-with-or-why-doesnt-it-return-an-fd-wit

Reply:
Well but, with the current code, that file is then moved to the final destination path so I can't use `TemporaryFile` as that gets deleted after it's closed/garbage collected. Maybe I could replace the two lines with a single statement (still two lines though):
```python
            with open tempfile.NamedTemporaryFile(suffix='.cover.jpg',
                                                  delete=True) as f:
                f.write(data)
```

BTW: I've used this style because it's used elsewhere in whipper's codebase.

Reply:
One thing I've just noticed is that the code is missing a `chmod` (to set the correct permissions) before moving the file.

Reply:
Did I miss something?

Reply:
Waiting for @Freso's reply here too.

Reply:
Or just download to memory.

Reply:
> Or just download to memory.

Didn't consider this possibility: in which location would you put the instructions to save the file?

Reply:
Done!